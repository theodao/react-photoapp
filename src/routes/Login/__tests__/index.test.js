import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Login, mapDispatchToProps, mapStateToProps } from '../';

describe('Testing login route', () => {
  afterEach(cleanup);

  it('Should match snapshot', () => {
    const { asFragment } = render(<Login auth={{ isLoggedIn: false }} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Testing mapState to props', () => {
    const state = {
      auth: {
        isLoggedIn: false,
      },
    };

    const mappedState = mapStateToProps(state);

    expect(mappedState.auth.isLoggedIn).toEqual(false);
  });

  it('Testing mapDispatchToProps', () => {
    const mockDispatchFunction = jest.fn();

    const mappedDistpachFunction = mapDispatchToProps(mockDispatchFunction);

    mappedDistpachFunction.dispatchLogin();
    expect(mockDispatchFunction).toHaveBeenCalled();
  });
});
