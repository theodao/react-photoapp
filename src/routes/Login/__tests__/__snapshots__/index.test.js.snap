// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login route Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ oxXIZ"
  >
    <div
      class="sc-AxiKw EEztA"
    >
      <div
        class="sc-AxhCb hOEozc"
      >
        Login
      </div>
      <form>
        <div>
          <label
            class="FormLabel u-block u-marginBottomTiny"
            style="font-size: 0.9rem;"
          >
            Email
            <span
              class="error"
            >
               *
            </span>
          </label>
          <input
            class="FormInput u-block u-widthFull u-webkitScrollbar"
            name="email"
            required=""
            style="font-size: 0.9rem;"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-AxjAm cExXRh"
        />
        <div>
          <label
            class="FormLabel u-block u-marginBottomTiny"
            style="font-size: 0.9rem;"
          >
            Password
            <span
              class="error"
            >
               *
            </span>
          </label>
          <input
            class="FormInput u-block u-widthFull u-webkitScrollbar"
            name="password"
            required=""
            style="font-size: 0.9rem;"
            type="password"
            value=""
          />
        </div>
        <div
          class="sc-AxjAm cExXRh"
        />
        <button
          class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone is-disabled u-cursorNotAllow u-pointerEventsNone u-widthFull u-textUppercase"
          disabled=""
          style="margin-bottom: 10px; margin-top: 10px; font-size: 0.9rem;"
        >
          <span
            class="Button-label u-inlineBlock"
          >
            Log in
          </span>
        </button>
        <button
          class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone u-cursorPointer u-widthFull u-textUppercase"
          style="font-size: 0.9rem;"
        >
          <span
            class="Button-label u-inlineBlock"
          >
            Sign up
          </span>
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
